typescript/sdk/src/token/EvmERC20WarpModule.hardhat-test.ts

```diff
@@ -104,12 +116,77 @@ describe('EvmERC20WarpHyperlaneModule', async () => {
       TOKEN_DECIMALS,
     );
 
+    vaultFactory = new ERC4626Test__factory(signer);
+    vault = await vaultFactory.deploy(token.address, TOKEN_NAME, TOKEN_NAME);
+
     baseConfig = routerConfigMap[chain];
 
     mailbox = Mailbox__factory.connect(baseConfig.mailbox, signer);
     ismAddress = await mailbox.defaultIsm();
   });
 
+  const movableCollateralTypes = Object.values(TokenType).filter(
+    isMovableCollateralTokenType,
+  ) as MovableTokenType[];
+
+  const assertAllowedRebalancers = async (
+    evmERC20WarpModule: EvmERC20WarpModule,
+    expectedRebalancers: string[],
+  ) => {
+    const currentConfig = await evmERC20WarpModule.read();
+    const currentRebalancers = Array.from(
+      currentConfig.allowedRebalancers ?? [],
+    );
+
+    expect(currentRebalancers.length).to.equal(expectedRebalancers.length);
+    currentRebalancers.forEach(
+      (rebalancer, idx) =>
+        expect(eqAddress(rebalancer, expectedRebalancers[idx])).to.be.true,
+    );
+  };
+
+  const getMovableTokenConfig = (): Record<
+    MovableTokenType,
+    HypTokenRouterConfig
+  > => {
+    return {
+      [TokenType.collateral]: {
+        ...baseConfig,
+        type: TokenType.collateral,
+        token: token.address,
+      },
+      [TokenType.collateralVault]: {
+        ...baseConfig,
+        type: TokenType.collateralVault,
+        token: vault.address,
+      },
+      [TokenType.collateralVaultRebase]: {
+        ...baseConfig,
+        type: TokenType.collateralVaultRebase,
+        token: vault.address,
+      },
+      [TokenType.native]: {
+        ...baseConfig,
+        type: TokenType.native,
+      },
+      [TokenType.synthetic]: {
+        ...baseConfig,
+        type: TokenType.synthetic,
+        decimals: TOKEN_DECIMALS,
+        name: TOKEN_NAME,
+        symbol: TOKEN_NAME,
+      },
+      [TokenType.syntheticRebase]: {
+        ...baseConfig,
+        type: TokenType.syntheticRebase,
+        decimals: TOKEN_DECIMALS,
+        name: TOKEN_NAME,
+        symbol: TOKEN_NAME,
+        collateralChainName: chain,
+      },
+    };
+  };
```

Comment:
I dont really understand what this is doing

Reply:
creating test configs for each token type that is movable programmatically